Feat/dynamic color borders

habovh opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. The three fields below are mandatory. -->

## Summary

Following up my issue #30377 I decided to have a look myself and contribute.

On iOS, border colors using `PlatformColor` or `DynamicColorIOS` do not update on the fly when the system appearance updates.
When the component mounts, the color is correct for the current appearance, but a component unmout/remount is required in order to see the color changing after a system appearance change.

Fixes #30377

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

## Changelog

<!-- Help reviewers and the release process by writing your own changelog entry. For an example, see:
https://github.com/facebook/react-native/wiki/Changelog
-->

[Internal] [Added] - Added `PlatformColor` and `DynamicColorIOS` examples to RNTester

## Test Plan

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes the user interface. -->

I added 2 border examples, one using `PlatformColor` and the other using `DynamicColorIOS`. I recorded the following before/after videos showing the effect of my changes:

https://user-images.githubusercontent.com/4186230/110828711-9c5dd600-8297-11eb-8bc8-bdc9054b6b44.mov

https://user-images.githubusercontent.com/4186230/110828800-b4cdf080-8297-11eb-9d23-07f69dc3a702.mov

sallar commented:
I would love this to be merged...

danilobuerger commented:
Duplicate of #29728

birkir commented:
> Duplicate of #29728

This PR is passing all checks, the other one does't.

Any chance of getting this PR merged, allowing for cherry pick into 0.64.2 ?

charlesbdudley commented:
Hi @habovh, thanks for your contribution. It does look like #29728 has been merged. Your additions to RNTester are great, would you mind revising this PR to just include that?

habovh commented:
@charlesbdudley Sure thing! I'll take care of that before the end of next week ☺️

habovh commented:
@charlesbdudley I've removed the native code related to supporting dynamic and platform colors since #29728 has been merged and rebased my branch on `facebook:main`.
Only my two border examples remain, I think this is ready to merge!
